model-api/src/commonMain/kotlin/org/modelix/model/area/AreaWithMounts.kt

```diff
@@ -159,6 +161,11 @@ class AreaWithMounts(val rootArea: IArea, mounts: Map<INode, IArea>) : IArea {
             return node.getChildren(role).map { NodeWrapper(getMountedArea(it)?.getRoot() ?: it) }
         }
 
+        override fun replaceNode(concept: ConceptReference?): INode {
```

Comment:
Interesting. I didn't think about the NodeWrapper. Having the default NodeWrapper always implement IReplaceableNode removes the purpose of IReplaceableNode (i.e. being able to check if it is supported before calling it.)

`AreaWithMount` and `CompositArea` are deprecate so it should be fine.